### PR TITLE
Fix initial load issue

### PR DESCRIPTION
**Pull request by micahmo**
## Pull Request Description

This PR fixes an issue where there's not an easy way to retry loading the feed if it times out due to a network issue. Note that in addition to changing the `ErrorMessage` action from "Account Settings" to "Retry", I also made the action button disabled and show a spinner when it is pressed. I did this because the main page doesn't react to `AuthStatus.loading`, so there was no other indication that any action had been taken. I'm not sure exactly why that was, but I have a feeling it had something to do with the main page indicator and the feed page's indicator not quite aligning together, and most of the time only the latter is needed.

> Review without whitespace.

## Issue Being Fixed

Issue Number: #1283

## Screenshots / Recordings

The demo shows first an unsuccessful retry attempt, and then a successful one.

https://github.com/thunder-app/thunder/assets/7417301/754a0835-30a1-4b1a-b173-6f5e5f8ed04d

## Checklist

- [ ] Did you update CHANGELOG.md?
- [ ] Did you use localized strings where applicable?
- [ ] Did you add `semanticLabel`s where applicable for accessibility?

### CODE REVIEW

**Comment by micahmo** [comment]
Good call! I've made `ErrorMessage` more generic so it can take a list of actions. Let me know what you think!

https://github.com/thunder-app/thunder/assets/7417301/f61ebd18-79d5-469e-bd62-4fb221cd7283

**Comment by hjiangsu** [comment]
I just have one small UI tweak - can we make it so that the buttons are in separate rows? Also, I think making one button the "primary" action and the other a "secondary" action would help with overall UI/UX!

For the primary/secondary, I'd suggest the following:
- Primary button uses the current look (elevated) - this can be the "Retry" option
- Secondary button uses a `TextButton` - this one can be the "Account Settings" option

**Comment by micahmo** [comment]
> can we make it so that the buttons are in separate rows?

I actually tried this first and liked side-by-side better, but I'll put it back!

> Secondary button uses a TextButton - this one can be the "Account Settings" option

Alright, I went ahead and pushed this. It looks below. Is that ok?

| image |
| - |

File: lib/shared/error_message.dart [hunk 1]
```diff
@@ -4,10 +4,14 @@ import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 class ErrorMessage extends StatelessWidget {
   final String? title;
   final String? message;
-  final String? actionText;
-  final VoidCallback? action;
+  final List<({String text, void Function() action, bool loading})>? actions;
```
Comment (hjiangsu) [inline lib/shared/error_message.dart:7]: Interesting use of records!

**Review by hjiangsu — APPROVED** [review]
LGTM! Just double checking, this is ready to be merged?

**Comment by micahmo** [comment]
> this is ready to be merged?

If the last screenshot looks good to you, yes!

File: lib/shared/error_message.dart (hunk 1 above)
Comment (micahmo) [reply lib/shared/error_message.dart:7]: Yeah it's close to needing a class. 🫣

**Pull request merged**